src/Action.hs

```diff
@@ -468,12 +501,37 @@ markIdeaInResultPhase iid rv = do
 
 -- * Topic handling
 
-topicInRefinementTimedOut :: (ActionPersist m, ActionSendMail m) => AUID Topic -> m ()
+topicInRefinementTimedOut :: (ActionPhaseChange m) => AUID Topic -> m ()
 topicInRefinementTimedOut = topicTimeout RefinementPhaseTimeOut
 
-topicInVotingTimedOut :: (ActionPersist m, ActionSendMail m) => AUID Topic -> m ()
+topicInVotingTimedOut :: (ActionPhaseChange m) => AUID Topic -> m ()
 topicInVotingTimedOut = topicTimeout VotingPhaseTimeOut
 
+topicInVotingResetToJury
+    :: (ActionPhaseChange m) => AUID Topic -> m ()
+topicInVotingResetToJury tid = do
+    topic <- mquery $ findTopic tid
+    case topic ^. topicPhase of
+        PhaseVoting _ -> topicPhaseChange topic VotingPhaseSetbackToJuryPhase
+        _             -> pure ()
+
+-- * Admin activities
+
+-- | Make a topic timeout if the timeout is applicable.
+-- FIXME: Only admin can do that
```

Comment:
Again, to clarify: please don't think about authorization yet, we are still working on the happy flow.

The `ideaCapabilities` should change the shape of the pages, but not impose any real restrictions on an adversary.

I think that there are a lot of places in the code where we have `when (something about user ...) $ do ...` that should check for ideaCapabilities instead.  ideaCapabilities should in turn take *all* information into account that affects the shape of the page, *including* user, user role, user id.

Are there any objections to doing them at all?  I'm not insisting on doing these changes now, but I would like to keep the idea in mind.